runtime/interpreter/interpreter_expression.go

```diff
@@ -1142,7 +1183,50 @@ func (interpreter *Interpreter) VisitPathExpression(expression *ast.PathExpressi
 	)
 }
 
-func (interpreter *Interpreter) VisitAttachExpression(_ *ast.AttachExpression) Value {
-	// TODO: implement this
-	return nil
+func (interpreter *Interpreter) VisitAttachExpression(attachExpression *ast.AttachExpression) Value {
+
+	base, ok := interpreter.evalExpression(attachExpression.Base).(*CompositeValue)
+	// we enforce this in the checker
+	if !ok {
+		panic(errors.NewUnreachableError())
+	}
+
+	// the `super` value must be accessible during the attachment's constructor, but we cannot
+	// set it on the attachment's `CompositeValue` yet, because the value does not exist. Instead
+	// we directly declare `super` as a `&base` reference in the scope of the constructor's invocation,
+	// and remove it immediately afterwards.\
+	baseValue := NewEphemeralReferenceValue(
+		interpreter,
+		false,
+		base,
+		interpreter.MustSemaTypeOfValue(base).(*sema.CompositeType),
+	)
+	// technically this makes `super` available to the arguments to the attachment constructor, but we enforce statically
+	// that it cannot appear there, so it's ok
+	interpreter.declareVariable(sema.SuperIdentifier, baseValue)
+
+	attachment, ok := interpreter.VisitInvocationExpression(attachExpression.Attachment).(*CompositeValue)
+	interpreter.activations.Remove(sema.SuperIdentifier)
```

Comment:
I have since determined that this is actually not safe; if `base` is already in scope (i.e. if we are inside an attachment body) this will shadow the existing `base` value too early. Working on a new approach for this now.